mapper/tedge_mapper/src/main.rs

```diff
@@ -1,27 +1,41 @@
+use crate::az_mapper::AzureMapper;
+use crate::c8y_mapper::CumulocityMapper;
+use crate::component::TEdgeComponent;
 use crate::error::*;
-use crate::size_threshold::*;
-use clock::WallClock;
-use flockfile::{Flockfile, FlockfileError};
-use mqtt_client::{Client, Config};
 use std::path::PathBuf;
 use std::str::FromStr;
+use strum_macros::*;
 use tedge_config::*;
-use tracing::{debug_span, error, info, Instrument};
 
 mod az_converter;
 mod az_mapper;
 mod c8y_converter;
 mod c8y_mapper;
+mod component;
 mod converter;
 mod error;
 mod mapper;
 mod size_threshold;
 
-const APP_NAME_C8Y: &str = "tedge-mapper-c8y";
-const APP_NAME_AZ: &str = "tedge-mapper-az";
 const DEFAULT_LOG_LEVEL: &str = "warn";
 const TIME_FORMAT: &str = "%Y-%m-%dT%H:%M:%S%.3f%:z";
 
+#[derive(EnumString)]
+pub enum ComponentName {
+    #[strum(serialize = "az")]
+    Azure,
+
+    #[strum(serialize = "c8y")]
+    C8y,
+}
+
+fn lookup_component(component_name: &ComponentName) -> Box<dyn TEdgeComponent> {
```

Comment:
can use `ComponentName` instead of `&ComponentName` and make `ComponentName` Copy and Clone.

Reply:
I'm not against that.

Adding Copy and Clone isn't always the right thing to do for enums though, because they might contain other types that are not Copy / Clone.

We don't have that problem here, so can mark it as Copy & Clone, but I'm wondering why you are suggesting it. Is `&ComponentName` un-idiomatic?